kbase-extension/static/kbase/js/common/simpleLogViewer.js

```diff
@@ -0,0 +1,818 @@
+define([
+    'bluebird',
+    'common/html',
+    'common/jobCommMessages',
+    'common/jobs',
+    'common/looper',
+    'common/props',
+    'common/ui',
+    'util/developerMode',
+], (Promise, html, jcm, Jobs, Looper, Props, UI, devMode) => {
+    'use strict';
+
+    const t = html.tag,
+        div = t('div'),
+        button = t('button'),
+        span = t('span'),
+        p = t('p'),
+        cssBaseClass = 'kb-log',
+        logContentStandardClass = `${cssBaseClass}__content`,
+        logContentExpandedClass = `${logContentStandardClass}--expanded`,
+        LOG_CONTAINER = 'log-container', // container for the whole widget
+        LOG_PANEL = 'log-panel', // panel displayed by clicking on the "Logs" title
+        LOG_LINES = 'log-lines'; // numbered list of log lines
+
+    let logContentClass = logContentStandardClass;
+
+    /**
+     * The entrypoint to this widget. This creates the job log viewer and initializes it.
+     * Starting it is left as a lifecycle method for the caller.
+     *
+     * config options:
+     *  showHistory   {boolean} - whether job status lines should include history
+     *                            (queue and run times)
+     *  logPollInterval {int}   - time in ms for how frequently to check for new logs
+     *  developerMode {boolean} - whether or not to enable developer mode, which
+     *                            prints out extra debugging information
+     */
+
+    class JobLogViewerCore {
```

Comment:
created by picking out all the log parsing / generating code from the job log viewer and converting it to use the looper for repeated log requests and the job manager for general job status requests.

Reply:
DATAUP-497 did not update the job log viewer calling code, so currently you can only view one set of job logs per bulk import cell. I was annoyed that the job log viewer did not use the job manager, so this fixes that, along with ripping out the FSM stuff that adds extra complexity to the log viewer. The old log viewer still lives on it `util/jobLogViewer`, although the plan is to eventually replace that when the app cell is converted to using the job manager.

Reply:
Sounds great to me.